datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -3281,7 +3279,7 @@ mod tests {
             (
                 col("c3"),
                 NullableInterval::NotNull {
-                    values: Interval::make(Some(0_i64), Some(2_i64), (false, false)),
+                    values: Interval::make(Some(0_i64), Some(2_i64)).unwrap(),
```

Comment:
I found the presence of `Interval::make` and `Interval::try_new` confusing as they both do the same thing but looking at this code I wonder "why sometimes use make and sometimes try_new?". 

Would it be possible to remove all calls t `Interval::make` and instead call `Interval::try_new` (could be a follow on PR)

Reply:
`make` is a utility function for use in tests only. Using `try_new` in test functions balloons the line count 🙂 I am adding a comment in the docstring to make this clear.